### ISSUE TITLE
[question] how to install a specific package with Debug build type

### ISSUE THREAD

**Issue by shelper** [conan-io/conan#14459]
### What is your question?

So, here is what happens to me, I build a package, say `foo`, and i am using `foo` for my other programs, say `bar`. 

And i installed `foo` for `bar` by putting `foo` into conanfile.py of `bar` project. Now i am debugging `bar` and i want to step into function that is defined in `foo`.  So i specifiec `foo` to have buid_type of Debug in the conanfile.py of `foo` project. 

but seems this does not work, because when i run `conan create` for foo, it still creates release version.
so what i tried is i modified the conan profile so that it creates debug version of `foo` by `conan profile update settings.build_type=Debug default`. However, conan then tries to install all the dependencies of `foo` in debug version too, which is not what i want. 

basically my question is: How can i only install the debug version of a package, but not its dependencies?

### Have you read the CONTRIBUTING guide?

- [x] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @shelper 

Thanks for your question.

You can use per-package settings, like ``conan install . -s build_type=Release -s &:build_type=Debug``

the per-package setting uses ``pattern:setting=value``, and the ``&`` pattern means "the consumer", but you can use something like ``mypkgname*:build_type=Debug``

**Comment by shelper** [comment]
humm, seems it does not work, 
i have a package called `modelwrapper/1.2.1`, which corresponding to the `foo` package above, and in my `bar` project, i run:
`conan install .. -s build_type=Release -s modelwrapper/1.2.1@:build_type=Debug`

i got everything installed, but i still cannot debug into the functions in `modelwrapper`, and i then i checked the installed package in `~/.conan/data/modelwrapper/1.2.1/_/_/package/bc798dcde41f73995d627e00e79aa87d344df0a7`, i found the `conainfo.txt`, and in that file it still says build type is Debug...
```
[settings]
    arch=x86_64
    build_type=Release
```

btw, i am using conan 1.60.1

**Comment by memsharded** [comment]
This is not clear, how do you want to step into the dependencies if your current project is Release mode? I thought you wanted to opposite, having your current project in Debug while having the dependencies in Release. What is the use case of developing in Release mode your current package and the dependencies in Debug? The current project needs to be in Debug in order to debug it?

> and in that file it still says build type is Debug...

But the conaninfo shows Release. is this a typo? If it is Release, then the pattern didn't work. Try ``-s modelwrapper*:build_type=xxx``

**Comment by shelper** [comment]
sorry, i will try to explain it again, 

my project is called `standalone`, and i have a coanfile for standalone that has line:

```
    requires = (
        "modelwrapper/1.2.1",
    )
```
I run `conan install ..` in the build folder, and got `modelwrapper` installed. and then I build `standalone` in debug mode, i can debug `standalone` but cannot jump into functions defined in `modelwrapper`.

So i tried to run  `conan install .. -s build_type=Debug` , but it turns out this command tries to install all dependencies in Debug build, which is not what i want, i just want `modelwrapper` to be Debug build, as i just want to debug into `modelwrapper`, not other dependencies, I also only want Release build for the dependencies that `modelwrapper` relies on.

I tried your method `conan install .. -s build_type=Release -s modelwrapper/1.2.1@:build_type=Debug`, but seems it does not work, conan still installs release build for `modelwrapper`, as can be confirmed by the content in file `~/.conan/data/modelwrapper/1.2.1/_/_/package/bc798dcde41f73995d627e00e79aa87d344df0a7/conaninfo.txt` says `build_type=Release`.

 Hope this is clear and thanks again for the help!

**Comment by memsharded** [comment]
Thanks for the clarifications, I think I get it now:

> I run conan install .. in the build folder, and got modelwrapper installed. and then I build standalone in debug mode, i can debug standalone but cannot jump into functions defined in modelwrapper.

Then try:
```
conan install . -s build_type=Release -s modelwrapper*:build_type=Debug -s standalone*:build_type=Debug
```
Basically you can specify which packages you want in Debug mode and which ones in Release mode.


>  conan still installs release build for modelwrapper, as can be confirmed by the content in file ~/.conan/data/modelwrapper/1.2.1/_/_/package/bc798dcde41f73995d627e00e79aa87d344df0a7/conaninfo.txt says build_type=Release.

Yes, this shouldn't happen, not sure what could be failing. Maybe it is a limitation of Conan 1, but I think that also works in Conan 1. If you could try the above suggested command, and share the full output, then do the ``conan search modelwrapper/1.2.1@`` and share the full output that would help.

**Comment by shelper** [comment]
i dont know why, but this time it worked... 
i think i did exactly yesterday but it didnt work ...
anyway, thanks for the help 👍 , closing this issue

**Comment by memsharded** [comment]
Happy to help, glad to know it is working now.
